R/common.R

```diff
@@ -59,24 +59,27 @@ lavBootstrap <- function(fit, samples = 1000, standard = FALSE, typeStd = NULL,
 
   # we actually need the SEs from the bootstrap not the SEs from ML or some other estimator
   N <- nrow(fit@boot$coef)
+  P <- ncol(fit@boot$coef)
+  freePars <- which(fit@ParTable$free != 0)
 
   # we multiply the var by (n-1)/n because lavaan actually uses n for the variance instead of n-1
   if (!standard) {
     # for unstandardized
-    fit@ParTable$se[fit@ParTable$free != 0] <- apply(fit@boot$coef, 2, sd) * sqrt((N-1)/N)
+    fit@ParTable$se[freePars] <- apply(fit@boot$coef, 2, sd) * sqrt((N-1)/N)
```

Comment:
We have this situation
```r
P <- ncol(fit@boot$coef)
freePars <- which(fit@ParTable$free != 0)
```
so evidently `length(apply(fit@boot$coef, 2, sd)) == P`. So it must be that `length(freePars) == P`? Otherwise this can lead to an error.


Also, I'm not sure you need to multiply by `sqrt((N-1)/N)`. I could be wrong, but isn't it customary to use the sample variance of bootstrap samples?

Reply:
no the length of freePars is not always P. When I use the boot sample to get the SEs it is easiest to take the whole sample which is of length P containing also the constrained parameters as 0s. The freePars are only the unconstrained parameters. So when I replace the SEs I can do 1:P which fits perfectly.